test/smoke/dashCore.js

```diff
@@ -1,18 +1,114 @@
-describe('DashCore', () => {
-  describe('All nodes', () => {
-    // `subversion` from `getNetworkInfo` response.
-    it('should have correct network type');
-
-    // The number of blocks should be almost the same (-3/+3) and block hash of particular
-    // block height should be the same. Using `blocks` and `bestblockhash` from `GetBlockChainInfo`.
-    it('should propagate blocks');
+const RpcClient = require('@dashevo/dashd-rpc/promise');
+const SSH = require('simple-ssh');
+const fs = require('fs');
+
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
+
+describe('All nodes', () => {
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    describe(`DashCore ${nodeName}`, () => {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+
+      it('should have correct network type', async () => {
+        const networkInfo = await rpc.getNetworkInfo();
+        expect(networkInfo.error).to.be.equal(null);
```

Comment:
I thought RPC throws an error if can't request to DC instance?

Reply:
getNetworkInfo returns
```
{ result: 
   { version: 120400,
     subversion: '/Dash Core:0.12.4(devnet=devnet-example)/',
     protocolversion: 70211,
     localservices: '0000000000000005',
     localrelay: true,
     timeoffset: 0,
     networkactive: true,
     connections: 9,
     networks: [ [Object], [Object], [Object] ],
     relayfee: 0.00001,
     incrementalfee: 0.00001,
     localaddresses: [ [Object] ],
     warnings: 'This is a pre-release test build - use at your own risk - do not use for mining or merchant applications' },
  error: null,
  id: 13122 }
```
I only check that error is null here

Reply:
I see, ok. Then you need to check error for every JSON RPC call or use jayson client (what I recommend).